internal/provider/network_connection_models.go

```diff
@@ -0,0 +1,289 @@
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	resourceschema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type NetworkConnectionModel struct {
+	ID          types.String `tfsdk:"id"`
+	NetworkID   types.String `tfsdk:"network_id"`
+	Description types.String `tfsdk:"description"`
+
+	AWS    *awsNetworkConnectionInfo    `tfsdk:"aws"`
+	Google *googleNetworkConnectionInfo `tfsdk:"google"`
+}
+
+type awsNetworkConnectionInfo struct {
+	Peering *awsNetworkConnectionPeeringInfo `tfsdk:"peering"`
+}
+
+type awsNetworkConnectionPeeringInfo struct {
+	VPCID         types.String `tfsdk:"vpc_id"`
+	AccountID     types.String `tfsdk:"account_id"`
+	RegionID      types.String `tfsdk:"region_id"`
+	IPv4CIDRBlock types.String `tfsdk:"ipv4_cidr_block"`
+	IPv6CIDRBlock types.String `tfsdk:"ipv6_cidr_block"`
+
+	PeeringConnectionID  types.String `tfsdk:"peering_connection_id"`
+	ManagedIPv4CIDRBlock types.String `tfsdk:"managed_ipv4_cidr_block"`
+	ManagedIPv6CIDRBlock types.String `tfsdk:"managed_ipv6_cidr_block"`
+}
+
+type googleNetworkConnectionInfo struct {
+	Name           types.String `tfsdk:"name"`
+	PeerNetworkURL types.String `tfsdk:"peer_network_url"`
+
+	ManagedNetworkURL types.String `tfsdk:"managed_network_url"`
+}
+
+var (
+	networkConnectionResourceSchema = resourceschema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Network Connection resource",
+
+		Attributes: map[string]resourceschema.Attribute{
+			"id": resourceschema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "Network Connection identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"network_id": resourceschema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "Network identifier",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"description": resourceschema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "Description of network connection",
+				Default:             stringdefault.StaticString(""),
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"aws": resourceschema.SingleNestedAttribute{
+				Optional:            true,
+				MarkdownDescription: "AWS connection info",
+				Attributes: map[string]resourceschema.Attribute{
+					"peering": resourceschema.SingleNestedAttribute{
+						Optional:            true,
```

Comment:
Why is this field `Optinal: true`? It must be required

Reply:
Someday we will have TransitGateway option for AWS connection instead of Peering.
So there will be both attributes `peering` and `tgw` with `Optinal: true`.
I don't know, will it better to set here `Required`?
Validation is checking [here](https://github.com/doublecloud/terraform-provider-doublecloud/blob/07c6208093b519a504c8be89d6d7ca0a9f00389f/internal/provider/network_connection_models.go#L160) and preserve in [tests](https://github.com/doublecloud/terraform-provider-doublecloud/blob/07c6208093b519a504c8be89d6d7ca0a9f00389f/internal/provider/network_connection_resource_test.go#L68)

Reply:
I think we can mark it as required to current phase and change to `optional` when we implement `TransitGateway`.